aea/crypto/ethereum.py

```diff
@@ -238,3 +242,47 @@ def is_transaction_settled(self, tx_digest: str) -> bool:
         if tx_status is not None:
             is_successful = True
         return is_successful
+
+    def generate_tx_nonce(self, seller: Address, client: Address) -> str:
+        """
+        Generate a random str message.
+
+        :param seller: the address of the seller.
+        :param client: the address of the client.
+        :return: return the hash in hex.
+        """
+        time_stamp = int(time.time())
+        aggregate_hash = Web3.keccak(
+            b"".join(
+                [seller.encode(), client.encode(), time_stamp.to_bytes(32, "big"), ]
+            )
+        )
+        return aggregate_hash.hex()
+
+    def validate_transaction(
+        self,
+        tx_digest: str,
+        seller: Address,
+        client: Address,
+        tx_nonce: str,
+        amount: int,
+    ) -> bool:
+        """
+        Check whether a transaction is valid or not.
+
+        :param seller: the address of the seller.
+        :param client: the address of the client.
+        :param tx_nonce: the transaction nonce.
+        :param amount: the amount we expect to get from the transaction.
+        :param tx_digest: the transaction digest.
+
+        :return: True if the random_message is equals to tx['input']
+        """
+
+        tx = self._api.eth.getTransaction(tx_digest)
+        return (
```

Comment:
Just in terms of style, I'd assign the result to a variable called `result` or `is_valid` and then have the last line `return is_valid` to make it more readable